src/Microsoft.AspNet.SignalR.JS/jquery.signalR.hubs.js

```diff
@@ -194,14 +196,15 @@
                         }
 
                         // Remove the registration from the list
-                        callbackSpace.splice(i, 1);
+                        callbackSpace.splice(callbackIndex, 1);
 
                         // Check if there are any registrations left, if not we need to destroy it.
                         if (callbackSpace.length === 0) {
                             delete callbackMap[eventName];
                         }
                     }
-                } else if (!callback) { // Check if we're removing the whole event and we didn't error because of an invalid callback
+                }
+                else if (!callbackIdentity) { // Check if we're removing the whole event and we didn't error because of an invalid callback
```

Comment:
```suggestion
                } else if (!callbackIdentity) { // Check if we're removing the whole event and we didn't error because of an invalid callback
```